blob/driver/driver.go

```diff
@@ -22,6 +22,15 @@ import (
 	"time"
 )
 
+// ReaderOptions controls Reader behaviors.
+type ReaderOptions struct {
```

Comment:
OK, so drivers import the concrete package so they can call the constructor, like blob.NewBucket. Why couldn't they also get ReaderOptions from the concrete package, instead of having to duplicate it?

Reply:
The concrete type imports the driver package, so the driver package can't import the concrete type without being circular. It's true that driver implementations import both, but the `ReaderOptions` need to be part of the interface.

In the past we've considered using type aliases instead of duplication, but there are enough cases where the options vary somewhat, and sometimes the documentation differs as well ("thou shalt" for drivers, "it will" for concrete type), so we've settled on this.